### PR TITLE
Remove PSync-related code and Vision Processing application

### PR DESCRIPTION
**Pull request by pulsastrix**
This PR removes the remaining PSync-related code and moves the vision processing example applications into a separate repository.

### CODE REVIEW

**Review by JKRhb — COMMENTED** [review]
To fix the CI failure, the paths in the `format.make` file in the `cmake` need to be adjusted slightly.

File: cmake/format.cmake [hunk 1]
```diff
@@ -21,7 +21,7 @@
 # format`. To only check whether the codebase is properly formatted, run `make
 # check-format` instead.
 
-set(FORMATTING_SOURCES apps/**/*.cpp  apps/**/*.hpp tests/*.cpp include/**/*.hpp)
+set(FORMATTING_SOURCES examples/**/*.cpp  examples/**/*.hpp tests/*.cpp include/**/*.hpp)
```
Comment (JKRhb) [inline cmake/format.cmake:24]: Hmm, seems to me as if we need to make the path more precise here for the formatting to work.

```suggestion
set(FORMATTING_SOURCES examples/wordCount/**/*.cpp tests/*.cpp include/**/*.hpp)
```

Comment (pulsastrix) [reply cmake/format.cmake]: I found a somewhat more generic solution for this in the comments beneath https://stackoverflow.com/a/36046965, commit abd7a17a3aba48a77fc4d829d107e299eaed5180 should fix the issue.

(also, I think the way `FORMATTING_SOURCES` was set before did not actually include all source files, `**` seems to not be expanded as one would expect)

File: cmake/format.cmake [hunk 2]
```diff
@@ -21,7 +21,14 @@
...
 
-set(FORMATTING_SOURCES apps/**/*.cpp  apps/**/*.hpp tests/*.cpp include/**/*.hpp)
+# Find all source files for formatting, see the replies to https://stackoverflow.com/a/36046965
+# and https://www.labri.fr/perso/fleury/posts/programming/using-clang-tidy-and-clang-format.html.
+file(GLOB_RECURSE FORMATTING_SOURCES 
+  examples/*.[ch]pp examples/*.[CH]PP examples/*.[ch]xx examples/*.[CH]XX examples/*.cc 
+  examples/*.CC examples/*.hh examples/*.HH examples/*.[CHch]
+  include/*.[ch]pp include/*.[CH]PP include/*.[ch]xx include/*.[CH]XX include/*.cc 
+  include/*.CC include/*.hh include/*.HH include/*.[CHch]
+)
```
Comment (JKRhb) [inline cmake/format.cmake:31]: That looks good to me :) However, maybe we could limit the list of file extensions to the ones we currently used, i.e. `.hpp` and `.cpp`? Or would it be better to cover all/most possible variants?

```suggestion
file(GLOB_RECURSE FORMATTING_SOURCES 
  examples/*.[ch]pp include/*.[ch]pp
)
```

File: cmake/sources.cmake [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# Find all source files for formatting, see the replies to https://stackoverflow.com/a/36046965
+# and https://www.labri.fr/perso/fleury/posts/programming/using-clang-tidy-and-clang-format.html.
+file(GLOB_RECURSE ALL_SOURCES
+  examples/*.[chCH][pP][pP]  examples/*.[chCH][xX][xX] examples/*.[cC][cC]
+  examples/*.[hH][hH] examples/*.[CHch]
+  include/*.[chCH][pP][pP]  include/*.[chCH][xX][xX] include/*.[cC][cC]
+  include/*.[hH][hH] include/*.[CHch]
```
Comment (JKRhb) [inline cmake/sources.cmake:23]:
```suggestion
  examples/*.[chCH][pP][pP] examples/*.[chCH][xX][xX] examples/*.[cC][cC]
  examples/*.[hH][hH] examples/*.[CHch]
  include/*.[chCH][pP][pP] include/*.[chCH][xX][xX] include/*.[cC][cC]
  include/*.[hH][hH] include/*.[CHch]
```

**Review by JKRhb — DISMISSED** [review]

**Review by JKRhb — APPROVED** [review]
LGTM! :)

**Pull request merged**